src/tcache.c

```diff
@@ -710,16 +815,22 @@ tsd_tcache_data_init(tsd_t *tsd) {
 	tcache_t *tcache = tsd_tcachep_get_unsafe(tsd);
 
 	assert(cache_bin_still_zero_initialized(&tcache->bins[0]));
-	size_t alignment = tcache_bin_alloc_alignment;
-	size_t size = sz_sa2u(tcache_bin_alloc_size, alignment);
+	thread_tcache_max_and_nhbins_init(tsd);
+	unsigned thread_nhbins = tsd_thread_nhbins_get(tsd);
+	size_t size, alignment;
+	cache_bin_info_t tcache_bin_info[TCACHE_NBINS_MAX] = {0};
```

Comment:
How much stack space does this use? If it's not too high, add a comment about it.

Question: is it correct that we got rid of the global tcache_bin_info with this commit? Nice win if so.

Reply:
Yes, the global `tcache_bin_info` is now meaningless considering that each thread should have its own setting on bins.

Reply:
Let's also add a comment about the size not being too high.

Also does the zero initializer get any compiler warnings? You might need to do `{ {0} };` to avoid the warning.

Reply:
Sure. Didn't see any compiler warnings around it so I think it's fine to be as is?